pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -101,10 +102,17 @@ func (i *InfraProvider) Provision(ctx context.Context, dir string, parents asset
 		tfvarsAsset,
 	)
 
+	updateClusterIDs := len(capiManifestsAsset.CapiClusterIDs) == 0
+
 	// Collect cluster and non-machine-related infra manifests
 	// to be applied during the initial stage.
 	infraManifests := []client.Object{}
 	for _, m := range capiManifestsAsset.RuntimeFiles() {
+		// Check for cluster definition so that we can collect the names.
+		if updateClusterIDs && strings.Contains(m.Filename, "01_capi-cluster") {
+			capiManifestsAsset.CapiClusterIDs = append(capiManifestsAsset.CapiClusterIDs, m.Object.GetName())
+		}
```

Comment:
What about:
```suggestion
		// Check for cluster definition so that we can collect the names.
		if cluster, ok := m.Object.(*clusterv1.Cluster); ok && updateClusterIDs {
			capiManifestsAsset.CapiClusterIDs = append(capiManifestsAsset.CapiClusterIDs, cluster.GetName())
		}
```

Reply:
Let me try that real quick.  I thought that is what I tried and got a strange error.

Reply:
That seems to have done the trick.  Still not sure what I did differently before.  /shrug